examples/plot_adversarial_mitigation.py

```diff
@@ -0,0 +1,367 @@
+# Copyright (c) Fairlearn contributors.
+# Licensed under the MIT License.
+
+"""
+================================================
+Mitigating Fairness using Adversarial Mitigation
+================================================
+"""
+# %%
+# This notebook demonstrates our implementation of the technique *Mitigating*
+# *Unwanted Biases with Adversarial Learning* as proposed by
+# `Zhang et al. 2018 <https://dl.acm.org/doi/pdf/10.1145/3278721.3278779>`_.
+#
+# In short, the authors take the classic supervised learning setting in which
+# a predictor neural network is trained, and extend it with an adversarial
+# network that aims to predict the sensitive feature given these
+# predictions and possibly also given the true target. Then, they train the
+# predictor not only to minimize its own loss, but also minimize the predictive
+# ability of the adversarial. If this model converges properly,
+# the adversary will attain a loss equal to the entropy, so the adversary
+# can not
+# predict the sensitive features from the predictions.
```

Comment:
```suggestion
# can not predict the sensitive features from the predictions.
```

Reply:
I'm not so keen on prediciting things from the predictions. Should it be 'recover the sensitive features?'